### ISSUE TITLE
Bug: Some file/drive icons are displayed as a small icon

### ISSUE THREAD

**Issue by foxnoodles** [files-community/Files#14499]
### Description

![Screenshot 2024-01-20 112232](https://github.com/files-community/Files/assets/1949828/dda5a994-ef95-46b6-be43-38726739a1cb)

On the left how it is displayed, on the right how it should be displayed (Properties screen from Files app itself) 

The icon is set via autorun.inf

`[autorun]
icon = KODAK.ico`

### Steps To Reproduce

1. Change a drives icon using an AutoRun file and provide an icon that only contains a 128*128 icon size
2. Open Files and the icon has large margins

### Requirements

1. Solution shouldn't reduce the quality of the other icons
2. Solution must remove margin from the icon

### Files Version

3.1.1.0

### Windows Version

22621.2861

### Log File

no log for that.

**Comment by yaira2** [comment]
Is the issue the padding? What does it show in File Explorer?

**Comment by foxnoodles** [comment]
The issue is icon scale obviously  Left one should have the same scale as the right one. I don't man know man I though this is kinda obvious :D

**Comment by yaira2** [comment]
I was fairly sure that's what you meant but I wanted to confirm 🙂 

What does it show in File Explorer? I'm wondering if the custom icon you're using is missing the size that's used on the home page (256x256) so the system is returning a smaller icon.

**Comment by foxnoodles** [comment]
Haha all good. 

Ok that is weird because the properties window in files app shows the correct size. 

As for the windows explorer here is the screen:

![image_2024-01-23_004256570](https://github.com/files-community/Files/assets/1949828/0ff1a271-dc24-4832-aac4-195d993fe1da)

**Comment by Mantej-Singh** [comment]
I have the same problem with my custom icons for drives. 
Not a big deal for me, but it's a nice to have feature.
![Screenshot 2024-01-22 155942](https://github.com/files-community/Files/assets/19365795/f0242609-7c9b-4480-b54a-1bf197d25296)

**Comment by Josh65-2201** [comment]
A workaround for now is to include 256 x 256 version in the icon file you are using.

**Comment by Mantej-Singh** [comment]
I can do that, thanks.

**Comment by Mantej-Singh** [comment]
@Josh65-2201 
256 x 256 version is working fine.

![image](https://github.com/files-community/Files/assets/19365795/eecd4c99-2052-48fe-b6b1-1acd2c950d5c)

**Comment by yaira2** [comment]
According to the docs, icons are supposed to include 16x16, 32x32, 48x48, and 256x256 sizes https://learn.microsoft.com/en-us/windows/win32/uxguide/vis-icons?redirectedfrom=MSDN#size-requirements. File Explorer seems to have a different way of falling back to a smaller size and we can try to do that as well, but officially this is working as expected.

**Comment by cOborski** [comment]
I've been playing around with custom icons recently, so here are a few screenshots that might help someone out.

![Screenshot 2024-01-23 212401](https://github.com/files-community/Files/assets/5460210/c85f45ee-d307-4aab-a406-cbbd6bcce840)
Screenshot of mostly custom icons for shortcuts in Files 
Note: Waterfox's shortcut does _not_ have a custom icon

![Screenshot 2024-01-23 212625](https://github.com/files-community/Files/assets/5460210/204b755b-5e69-4403-b32a-3c511426fe3a)
Screenshot of Waterfox's executable file, including the preview pane

![Screenshot 2024-01-23 212813](https://github.com/files-community/Files/assets/5460210/6ffbfd6e-004b-4613-b129-87980d671836)
Screenshot of mostly custom shortcut in Windows 11 File Explorer

Note:
- The custom icons that appear small in Files are 128x128
- Waterfox's icon appears differently depending on whether it's the executable or a shortcut

**Comment by foxnoodles** [comment]
> According to the docs, icons are supposed to include 16x16, 32x32, 48x48, and 256x256 sizes https://learn.microsoft.com/en-us/windows/win32/uxguide/vis-icons?redirectedfrom=MSDN#size-requirements. File Explorer seems to have a different way of falling back to a smaller size and we can try to do that as well, but officially this is working as expected.

Again, the properties window in FILES app already does that (as shown in the very first image in this thread), it fills the container with whichever biggest size is available. Let's just do the same for drive widgets.

**Comment by foxnoodles** [comment]
wdym?  In my particular case the "Kodak" icon has only 128x128px size variant. There are no smaller versions in that ico file. And the properties window icon does not look blurry at all. Explorer icon looks slightly blurry as it is upscaled obviously but still not too bad. I don't know.

**Comment by yaira2** [comment]
If we copied what we have in the Properties window, there would be a drop in the icon quality on the home page.

**Before**
![image](https://github.com/files-community/Files/assets/39923744/7876b347-4e9d-4ec1-a7fd-42a461d587cf)

**After**
![image](https://github.com/files-community/Files/assets/39923744/c2010e4f-9241-4dfc-8a35-0a746366ca2b)

**Comment by foxnoodles** [comment]
That's prolly on hi-dpi screen. Mine is regular 96 dpi. and homepage icon container is 128x128 in size so it does not look blurry at all. 
anyways, is there a way to just check for the highest available quality and use that? Because clearly the home page does not show my 128x128px icon correctly. it shows 16x16px one which is not even present in the ico file.

**Comment by Josh65-2201** [comment]
People do use high dpi so it isn't a good solution, If make your icon 256 x 256 resolution it will show normally then.

**Comment by yaira2** [comment]
> That's prolly on hi-dpi screen. Mine is regular 96 dpi. and homepage icon container is 128x128 in size so it does not look blurry at all. anyways, is there a way to just check for the highest available quality and use that? Because clearly the home page does not show my 128x128px icon correctly. it shows 16x16px one which is not even present in the ico file.

That's what we proposed here https://github.com/files-community/Files/issues/14499#issuecomment-1906660032 but we'll continue looking to find the best solution.

**Comment by foxnoodles** [comment]
So it is working as expected by scaling down 128x128px icon to 16x16px and displaying that inside 128x128px container?

**Comment by yaira2** [comment]
There is a potential solution in #1547 but we're open to other ideas as well.

**Comment by cOborski** [comment]
Making the icon 256x256 resolution might not necessarily fix the problem. For example:

![Screenshot 2024-01-24 203319](https://github.com/files-community/Files/assets/5460210/5a4f51fb-9ace-4227-9fc0-de952b753d3c)
Waterfox only has a 256x256 resolution icon

![Screenshot 2024-01-24 203355](https://github.com/files-community/Files/assets/5460210/5bcbe758-26fb-4a6c-831a-45f20aedfe6c)
Obsidian has 16x16 through 256x256

I don't think the problem is simply _the icon isn't big enough_, otherwise Waterfox's icon in my previous screenshot would be displayed the same size as Obsidian's icon.

**Comment by yaira2** [comment]
@foxnoodles this issue is mainly focusing on the Home Page but I'll be looking at the Grid View layout once this is complete.

**Comment by foxnoodles** [comment]
> There is a potential solution in #1547 but we're open to other ideas as well.

I think you linked the wrong issue.

**Comment by yaira2** [comment]
A big shoutout to everyone who commented on this issue as well as @Josh65-2201, @mdtauk and @ahmed605 for their help on solving this issue. The fix has been added in #14779 and will be included in the next release.
